src/tests/dotnet-trace/ChildProcessTests.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.NETCore.Client;
+using System;
+using Xunit;
+using Xunit.Abstractions;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+
+    public class ChildProcessTests
+    {
+        // Pass ITestOutputHelper into the test class, which xunit provides per-test
+        public ChildProcessTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+        }
+
+        private ITestOutputHelper OutputHelper { get; }
+
+        private void LaunchDotNetTrace(string command, out int exitCode, out string stdOut, out string stdErr)
+        {
+            string dotnetTracePathWithArgs = CommonHelper.GetTraceePathWithArgs(traceeName: "dotnet-trace").Replace("net5.0", "netcoreapp2.1");
+            ProcessStartInfo startInfo = new ProcessStartInfo(CommonHelper.HostExe, $"{dotnetTracePathWithArgs} {command}");
+
+            OutputHelper.WriteLine($"Launching: {startInfo.FileName} {startInfo.Arguments}");
+            startInfo.RedirectStandardInput = true;
+            startInfo.RedirectStandardError = true;
+            startInfo.RedirectStandardOutput = true;
```

Comment:
Is it possible to use the System.CommandLine APIs to achieve this behavior?

e.g., https://github.com/dotnet/diagnostics/blob/89a8ceb035c6bb20e59d5b1f2be106338198b7d0/src/tests/dotnet-stack/StackTests.cs#L37-L53

Reply:
No, the design of the `--show-child-io` is specifically engineered to avoid any redirection of the IO of the child process, and thus this sort of test will not work correctly.